apm-agent-plugins/apm-jaxrs-plugin/src/test/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionNameInstrumentationTest.java

```diff
@@ -106,6 +106,20 @@ public void testJaxRsTransactionNameWithJaxrsAnnotationInheritance() {
         assertThat(actualTransactions.get(2).getName().toString()).isEqualTo("ResourceWithPathOnAbstract#testMethod");
     }
 
+    @Test
+    public void testProxyClassInstrumentationExclusion() {
+        when(config.getConfig(JaxRsConfiguration.class).isEnableJaxrsAnnotationInheritance()).thenReturn(true);
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install());
+
+        doRequest("testViewProxy");
+        doRequest("testProxyProxy");
+
+        List<Transaction> actualTransactions = reporter.getTransactions();
+        assertThat(actualTransactions).hasSize(2);
```

Comment:
Why do we capture them at all? Through a base servlet or servlet filter?

Reply:
I guess these are just EJB proxies for the actual resource class. But maybe we should rather exclude those proxies globally here:

https://github.com/elastic/apm-agent-java/blob/4ab2b6f358eb040df407d830831d9d03a07eef67/apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java#L229-L246

by adding `WildcardMatcher.caseSensitiveMatcher("*$Proxy")` and `WildcardMatcher.caseSensitiveMatcher("*$view")`

Reply:
I meant- why are they captured in the test?

Reply:
In a real-world scenario, the transaction would be created by the servlet instrumentation. The JAX-RS instrumentation does not create transactions, it only sets the name. In this test, the transaction is manually created by the private `doRequest` method.

WDYT about moving it to the default `classes_excluded_from_instrumentation`?

Reply:
Ahh, missed the manual transaction creation.

I am not sure about moving to `classes_excluded_from_instrumentation`. Here it causes a transaction name explosion. Maybe in some scenarios it will capture things that otherwise won't be captured? Like some implementation that creates `$Proxy` implementations to interface stubs without a resource class in between? If you don't think this is a valid concern, go ahead and move it there.